packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2104,3 +2139,165 @@ class InputDecoration {
+  ///
+  /// If null, defaults to a value derived from the base [TextStyle] for the
+  /// input field and the current [Theme].
+  final TextStyle hintStyle;
+
+  /// The style to use for the [InputDecoration.errorText].
+  ///
+  /// If null, defaults of a value derived from the base [TextStyle] for the
+  /// input field and the current [Theme].
+  final TextStyle errorStyle;
+
+  /// Whether the input decorator's child is part of a dense form (i.e., uses
+  /// less vertical space).
+  ///
+  /// Defaults to false.
+  final bool isDense;
+
+  /// The padding for the input decoration's container.
+  ///
+  /// The decoration's container is the area which is filled if
+  /// [InputDecoration.isFilled] is true and bordered per the [border].
+  /// It's the area adjacent to [InputDecoration.icon] and above the
+  /// [InputDecoration.icon] and above the widgets that contain
+  /// [InputDecoration.helperText], [InputDecoration.errorText], and
+  /// [InputDecoration.counterText].
+  ///
+  /// By default the `contentPadding` reflects [isDense] and the type of the
+  /// [border]. If [isCollapsed] is true then `contentPadding` is
+  /// [EdgeInsets.zero].
+  final EdgeInsets contentPadding;
+
+  /// Whether the decoration is the same size as the input field.
+  ///
+  /// A collapsed decoration cannot have [InputDecoration.labelText],
+  /// [InputDecoration.errorText], or an [InputDecoration.icon].
+  final bool isCollapsed;
+
+  /// The style to use for the [InputDecoration.prefixText].
+  ///
+  /// If null, defaults to the [hintStyle].
+  final TextStyle prefixStyle;
+
+  /// The style to use for the [InputDecoration.suffixText].
+  ///
+  /// If null, defaults to the [hintStyle].
+  final TextStyle suffixStyle;
+
+  /// The style to use for the [InputDecoration.counterText].
+  ///
+  /// If null, defaults to the [helperStyle].
+  final TextStyle counterStyle;
+
+  /// If true the decoration's container is filled with [fillColor].
+  ///
+  /// Typically this field set to true if [border] is
+  /// [const UnderlineInputBorder()].
+  ///
+  /// The decoration's container is the area which is filled if [isFilled] is
+  /// true and bordered per the [border]. It's the area adjacent to
+  /// [InputDecoration.icon] and above the widgets that contain
+  /// [InputDecoration.helperText], [InputDecoration.errorText], and
+  /// [InputDecoration.counterText].
+  ///
+  /// This property is false by default.
+  final bool filled;
+
+  /// The color to fill the decoration's container with, if [filled] is true.
+  ///
+  /// By default the fillColor is based on the current [Theme].
+  ///
+  /// The decoration's container is the area which is filled if [isFilled] is
+  /// true and bordered per the [border]. It's the area adjacent to
+  /// [InputDecoration.icon] and above the widgets that contain
+  /// [InputDecoration.helperText], [InputDecoration.errorText], and
+  /// [InputDecoration.counterText].
+  final Color fillColor;
```

Comment:
filled and fillColor should mention that the shape of the fill area comes from border